src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -887,6 +888,43 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
     return vm
 
 
+def auto_shutdown_vm(cmd, resource_group_name, vm_name, off=None, email=None, webhook=None, time=None,
+                     location=None):
+    from msrestazure.tools import resource_id
+    from azure.mgmt.devtestlabs.models import Schedule
+    client = _dev_test_labs_client_factory(cmd.cli_ctx)
+    name = 'shutdown-computevm-' + vm_name
+    vm_id = resource_id(subscription=client.config.subscription_id, resource_group=resource_group_name,
+                        namespace='Microsoft.Compute', type='virtualMachines', name=vm_name)
+    if off:
+        if email is not None or webhook is not None or time is not None:
+            # I don't want to disturb users. So I warn instead of raising an error.
+            logger.warning('If --off, other parameters will be ignored.')
+        return client.global_schedules.delete(resource_group_name, name)
+
+    if time is None:
+        raise CLIError('usage error: --time is a required parameter')
+    if email is not None and webhook is None:
+        raise CLIError('usage error: --webhook is missing')
+    daily_recurrence = {'time': time}
+    notification_settings = None
+    if webhook:
+        notification_settings = {
+            'emailRecipient': email,
+            'webhookUrl': webhook,
+            'timeInMinutes': 30,
+            'status': 'Enabled'
+        }
+    schedule = Schedule(status='Enabled',
+                        target_resource_id=vm_id,
+                        daily_recurrence=daily_recurrence,
+                        notification_settings=notification_settings,
+                        time_zone_id='UTC',
+                        task_type='ComputeVmShutdownTask',
+                        location=location)
+    return client.global_schedules.create_or_update(resource_group_name, name, schedule)
```

Comment:
why this using devtestlab client? if this VM is not created inside a lab, does the feature work?

Reply:
Yes, it works. I debug this feature in portal. I implement it in the same way. It's a global schedule that can target VM outside lab.